map_gui/src/load.rs

```diff
@@ -247,9 +258,30 @@ mod wasm_loader {
                     Ok(resp_value) => {
                         let resp: Response = resp_value.dyn_into().unwrap();
                         if resp.ok() {
-                            let buf = JsFuture::from(resp.array_buffer().unwrap()).await.unwrap();
-                            let array = js_sys::Uint8Array::new(&buf);
-                            tx.send(Ok(array.to_vec())).unwrap();
+                            let total_bytes = resp
+                                .headers()
+                                .get("Content-Length")
+                                .unwrap()
+                                .unwrap()
+                                .parse::<usize>()
+                                .unwrap();
+                            tx_total_bytes.send(total_bytes).unwrap();
+
+                            let raw_body = resp.body().unwrap_throw();
+                            let body = ReadableStream::from_raw(raw_body.dyn_into().unwrap_throw());
+                            let mut stream = body.into_stream();
+                            let mut buffer = Vec::new();
+                            while let Some(Ok(chunk)) = stream.next().await {
+                                let array = js_sys::Uint8Array::new(&chunk);
+                                if let Err(err) =
+                                    tx_read_bytes.try_send(array.byte_length() as usize)
+                                {
+                                    warn!("Couldn't send update on bytes: {}", err);
+                                }
+                                // TODO Can we avoid this clone?
```

Comment:
There seem to be browser APIs for concatenating buffers, so there might be a more efficient way to do this. I haven't figured it out yet.

One idea I might play with eventually is to just have a single channel, and stream the chunks back directly. All the byte counting could happen in `event`.